### PR TITLE
Add new variable tidb_opt_three_stage_distinct_agg

### PR DESCRIPTION
**Pull request by fixdb**
### What is changed, added or deleted? (Required)

Added new variable `tidb_opt_three_stage_distinct_agg`.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -2119,6 +2119,14 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 - When the value is set to `OFF`, the non-transactional DML statement stops immediately at the first error and returns the error. All the following batches are canceled.
 - When the value is set to `ON` and an error occurs in a batch, the following batches will continue to be executed until all batches are executed. All errors occurred during the execution process are returned together in the result.
 
+### tidb_opt_3stage_distinct_agg <span class="version-mark">New in v6.3.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Default value: `ON`
+- This variable specifies whether to rewrite a `COUNT(DISTINCT)` aggregate into 3 stages aggregation in MPP mode.
+- This variable currently applies to aggregation with only 1 `COUNT(DISTINCT)`.
```
Comment (shichun-0415) [inline system-variables.md:2128]:
```suggestion
- This variable currently applies to aggregations with only one `COUNT(DISTINCT)`.
```

File: system-variables.md [hunk 2]
```diff
@@ -2119,6 +2119,14 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 - When the value is set to `OFF`, the non-transactional DML statement stops immediately at the first error and returns the error. All the following batches are canceled.
 - When the value is set to `ON` and an error occurs in a batch, the following batches will continue to be executed until all batches are executed. All errors occurred during the execution process are returned together in the result.
 
+### tidb_opt_3stage_distinct_agg <span class="version-mark">New in v6.3.0</span>
```
Comment (shichun-0415) [inline system-variables.md:2122]: 变量名还能改吗？3stage 连在一起不太清晰，建议 3-stage 或 three-stage

Comment (fixdb) [reply system-variables.md:2122]: 可以的。那就 `tidb_opt_multi_stage_distinct_agg`吧，考虑到以后可能不仅仅局限于3阶段。

Comment (shichun-0415) [reply system-variables.md:2122]: LGTM

Comment (shichun-0415) [inline system-variables.md:2122]:
```suggestion
### tidb_opt_three_stage_distinct_agg <span class="version-mark">New in v6.3.0</span>
```

File: system-variables.md [hunk 3]
```diff
@@ -2381,6 +2381,14 @@ explain select * from t where age=5;
 - Default value: `OFF`
 - This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation column has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance.
 
+### tidb_opt_three_stage_distinct_agg <span class="version-mark">New in v6.3.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
```
Comment (AilinKid) [inline system-variables.md:2387]: YES

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

File: system-variables.md (hunk 3 above)
Comment (fixdb) [reply system-variables.md:2387]: @shichun-0415 There are multiple uses: YES vs Yes.
Which one is preferred?

Comment (shichun-0415) [reply system-variables.md:2387]: `Yes` is better. I will change `YES` to `Yes`.

Comment (shichun-0415) [reply system-variables.md:2387]: Fixed in ddb9f3e

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: system-variables.md [hunk 4]
```diff
@@ -2381,6 +2381,14 @@ explain select * from t where age=5;
...
+- Default value: `ON`
```
Comment (shichun-0415) [inline system-variables.md:2388]:
```suggestion
- Type: Boolean
- Default value: `ON`
```

File: system-variables.md [hunk 5]
```diff
@@ -2381,6 +2381,15 @@ explain select * from t where age=5;
 - Default value: `OFF`
 - This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation column has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance.
 
+### tidb_opt_three_stage_distinct_agg <span class="version-mark">New in v6.3.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Type: Boolean
+- Default value: `ON`
+- This variable specifies whether to rewrite a `COUNT(DISTINCT)` aggregation into a three-stage aggregation in MPP mode.
```
Comment (TomShawn) [inline system-variables.md:2390]:
```suggestion
- This variable specifies whether to rewrite a `COUNT(DISTINCT)` aggregation condition in a query into a three-stage aggregation in the MPP mode.
```

File: system-variables.md [hunk 6]
```diff
@@ -2381,6 +2381,15 @@ explain select * from t where age=5;
...
+- This variable currently applies to aggregations with only one `COUNT(DISTINCT)`.
```
Comment (TomShawn) [inline system-variables.md:2391]: This is a confusing. Does it mean:
```suggestion
- This variable currently applies to the aggregation that only contains one `COUNT(DISTINCT)`.
```

**Review by TomShawn — APPROVED** [review]
rest LGTM

**Pull request merged**